src/sqlfluff/rules/L047.py

```diff
@@ -87,7 +93,64 @@ def _eval(self, segment, **kwargs):
                         ),
                     ],
                 )
-            if not self.prefer_count_1 and f_content[0].is_type("expression"):
+            elif self.prefer_count_1 and f_content[0].is_type("expression"):
```

Comment:
Looking at this code, it seems pretty repetitive now -- i.e. the code for `count(0)` and `count(1)` is often identical. Do you think we could DRY this up (i.e. https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)?

Reply:
You were right. This could be simplified - a lot!